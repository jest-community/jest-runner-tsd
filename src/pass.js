const { toTestResult } = require('./toTestResult');

/**
 * @typedef {object} Test
 * @property {string} path
 * @property {string} title
 */

/**
 * @typedef {object} Options
 * @property {number} start
 * @property {number} end
 * @property {Test} test
 * @property {number} numPassed
 */

/**
 * @param {Options} options
 */
module.exports.pass = ({ start, end, test, numPassed }) => {
  return toTestResult({
    stats: {
      failures: 0,
      passes: numPassed,
      pending: 0,
      todo: 0,
      start,
      end,
    },
    skipped: false,
    tests: [{ duration: end - start, ...test }],
    jestTestPath: test.path,
  });
};
