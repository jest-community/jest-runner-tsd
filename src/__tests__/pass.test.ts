import { expect, test } from '@jest/globals';
import { pass } from '../pass';

test('pass', () => {
  const testResult = pass({
    start: 0,
    end: 1200,
    test: {
      path: '/path/to/e2e/__fixtures__/js-passing/index.test.ts',
      title: 'tsd typecheck',
    },
    numPassed: 5,
  });

  expect(testResult).toMatchSnapshot();
});
