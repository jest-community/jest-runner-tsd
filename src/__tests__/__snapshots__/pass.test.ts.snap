// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass 1`] = `
Object {
  "console": null,
  "failureMessage": undefined,
  "numFailingTests": 0,
  "numPassingTests": 5,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "perfStats": Object {
    "end": 1200,
    "start": 0,
  },
  "skipped": undefined,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "/path/to/e2e/__fixtures__/js-passing/index.test.ts",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 1200,
      "failureMessages": Array [],
      "fullName": undefined,
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "tsd typecheck",
    },
  ],
}
`;
